packages/svelte/src/internal/client/reactivity/effects.js

```diff
@@ -574,14 +574,15 @@ export function resume_effect(effect) {
  */
 function resume_children(effect, local) {
 	if ((effect.f & INERT) === 0) return;
-	effect.f ^= INERT;
 
 	// If a dependency of this effect changed while it was paused,
 	// apply the change now
 	if (check_dirtiness(effect)) {
 		update_effect(effect);
 	}
 
+	effect.f ^= INERT;
```

Comment:
seems like the order is important? Is there a small comment that we can add to explain it?